docs/usage/models.md

```diff
@@ -158,6 +159,46 @@ print(m.model_dump())
 
 For self-referencing models, see [postponed annotations](postponed_annotations.md#self-referencing-or-recursive-models).
 
+## Rebuild model schema
+
+The model schema can be rebuilt using `model_rebuild()`. This is useful for building recursive generic models.
```

Comment:
We need to tell users when it's useful. Something like "we try to determine when this is necessary and error if it wasn't done but you may want to call model_rebuild() proactively when dealing with recursive and/or generics.